koku/api/user_access/view.py

```diff
@@ -164,26 +131,14 @@ def get(self, request, **kwargs):
         if flag.lower() == "true" and not beta:
             return Response({"data": False})
 
-        source_type = query_params.get("type")
-        if source_type:
-            source_accessor = next(
-                (item for item in self._source_types if item.get("type") == source_type.lower()), False
-            )
-            if source_accessor:
-                access_class = source_accessor.get("access_class")
-                pre_release_feature = source_accessor.get("pre_release_feature")
-                access_granted = access_class(user_access).access(admin_user, pre_release_feature)
-                return Response({"data": access_granted})
-            else:
-                return Response({f"Unknown source type: {source_type}"}, status=status.HTTP_400_BAD_REQUEST)
-
-        data = []
-        for source_type in self._source_types:
-            access_granted = False
-            access_granted = source_type.get("access_class")(user_access).access(
-                admin_user, source_type.get("pre_release_feature")
-            )
-            data.append({"type": source_type.get("type"), "access": access_granted})
+        ui_feature_access = UIFeatureAccess(user_access, admin_user)
+        is_valid_param = ui_feature_access.set_access_key_list(query_params.get("type"))
```

Comment:
I think the naming convention here is confusing. I think we need a validating method that determines if `query_params.get("type")` is valid. I think this `set_access_key_list` method is trying to be too clever.

you could define:
```
def validate_type(self, query_param):
   if not query_param:
      return True
   return query_param in self.access_key_mapping
```
which can then be used:
```
ui_feature_access = UIFeatureAccess(user_access, admin_user)
if param_type := query_params.get("type"):
   if not ui_feature_access.validate_type(param_type)
       return Response({f"Unknown source type: {query_params.get('type')}"}, status=status.HTTP_400_BAD_REQUEST)
   ui_feature_access.update_access_key_list(param_type)
...
```